modules/lang-painless/src/main/java/org/elasticsearch/painless/MethodWriter.java

```diff
@@ -262,6 +262,50 @@ public void writeToStrings() {
         }
     }
 
+    /** Writes a dynamic binary instruction: returnType, lhs, and rhs can be different */
+    public void writeDynamicBinaryInstruction(Location location, Type returnType, Type lhs, Type rhs, Operation operation) {
+        org.objectweb.asm.Type descriptor = org.objectweb.asm.Type.getMethodType(returnType.type, lhs.type, rhs.type);
```

Comment:
Also the Operation enum we have is currently not helpful here to address these issues, because it is more of a syntax level thing. It only has one `-` for example. It does not know about any semantics.